Code/GraphMol/DetermineBonds/DetermineBonds.cpp

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright (C) 2022 Sreya Gogineni and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "DetermineBonds.h"
+#include <GraphMol/RDKitBase.h>
+#include <YAeHMOP/EHTTools.h>
+#include <iostream>
+
+namespace RDKit {
+
+void connectivityHuckel(RWMol &mol, int charge) {
+    auto numAtoms = mol.getNumAtoms();
+    mol.getAtomWithIdx(0)->setFormalCharge(charge);
+    EHTTools::EHTResults res;
+    bool success = runMol(mol, res);
+    double *mat = res.reducedOverlapPopulationMatrix.get();
+    int matInd = 0;
+    int range = 1;
+    for (unsigned int i = 0; i < numAtoms; i++) {
+        for (unsigned int j = 0; j < range; j++) {
+            if (mat[matInd] >= 0.15 && i != j) {
```

Comment:
The more usual way to do this is to do the faster comparison first:
```suggestion
            if (i != j && mat[matInd] >= 0.15 ) {
```